vars/publishSmokeTestResults.groovy

```diff
@@ -0,0 +1,256 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+/** Library to fetch the failing Integration test details at the end of Integration Test Jenkins build and index the results to OpenSearch Metrics cluster.
+ *
+ * @param Map args = [:] args A map of the following parameters.
+ * @param args.distributionBuildUrl <required> - The jenkins distribution build number.
+ * @param args.testReportManifestYml <required> - The generated test report YAML file using test report workflow.
+ * @param args.jobName <required> - The smoke test job name, used in `testReportManifestYmlUrl`.
+ */
+
+import groovy.json.JsonOutput
+import java.text.SimpleDateFormat
+import java.util.Date
+
+void call(Map args = [:]) {
+
+    def isNullOrEmpty = { str ->
+        str == null || (str instanceof String && str.trim().isEmpty())
+    }
+    // Check if any args is equals to null or it is a test run
+    if (isNullOrEmpty(args.distributionBuildUrl) || isNullOrEmpty(args.testReportManifestYml) || isNullOrEmpty(args.jobName) || args.jobName.equals('dummy_job')) {
+        return null
+    }
```

Comment:
Maybe separate into a different method such as https://github.com/opensearch-project/opensearch-build-libraries/blob/main/vars/checkRequestAssignReleaseOwners.groovy#L66-L82 that would throw right error about what is missing. Also would help adding tests for the same.

Reply:
Would we want the entire job to fail if the indexing step doesn't complete successfully?

From my perspective, this library serves as an add-on to the existing smoke tests and is primarily responsible for ingesting results into the metrics cluster. If the ingestion fails, it shouldn’t impact the core test execution or its results.

This seems consistent with how we handle indexing in the integration tests as well, as shown here:
https://github.com/opensearch-project/opensearch-build-libraries/blob/91a1c453f4e14731d164de2e0feb4d4273b28487/vars/publishIntegTestResults.groovy#L25-L32

Reply:
I see what you mean. Do we have guard rails in place that wrong results are not being indexed? Such as empty values or nulls? 
Reason being if we are taking after actions based on those data, we need to make sure we have the correct data.